request_test.go

```diff
@@ -843,7 +843,7 @@ func TestRequest_BodyChunked(t *testing.T) {
 	assert.Equal(t, "METHOD", client.req.Method)
 	assert.Equal(t, "url", client.req.URL.String())
 	assert.Equal(t, make(http.Header), client.req.Header)
-	assert.Equal(t, "body", string(resp.content))
+	assert.Equal(t, "body", string(resp.getContent(resp.chain)))
```

Comment:
`resp.getContent(resp.chain)` is not quite correct usage.

getContent may call chain.fail, and chain.fail requires to be called between chain.enter / chain.leave, i.e. inside an operation. If you try to call it outside of operation, it will panic instead of reporting failure.

I suggest to replace:

```go
string(resp.getContent(resp.chain))
```

with:

```go
resp.Body().Raw()
```

and replace:

```go
bytes.NewReader(resp.getContent(resp.chain))
```

with:

```go
strings.NewReader(resp.Body().Raw())
```